spec/index.bs

```diff
@@ -3300,9 +3301,19 @@ GPUCommandBuffer includes GPUObjectBase;
 
 <script type=idl>
 dictionary GPUCommandBufferDescriptor : GPUObjectDescriptorBase {
+    boolean executionTimeEnabled = false;
```

Comment:
This flag should be used during `GPUCommandEncoder.finish()` to create `[=a new promise=]` and store it in an internal slot on the `GPUCommandBuffer` if measureExecutionTime was true.

The promise should then be resolved in `GPUQueue.submit().`

OK to just leave `Issue:`s in the places where we need to define some of this if needed, but I think we have the places to do these now.

Reply:
> The promise should then be resolved in GPUQueue.submit().

After `GPUQueue.submit()`, the results will be resolved to GPUBuffer. Shouldn't the executionTime promise be resolved when we call GPUCommandBuffer.executionTime  which read the result to CPU memory from GPUBuffer?

Reply:
Sorry, you're right, it shouldn't resolve *inside* GPUQueue.submit(). But I think (since we have the measureExecutionTime flag) we should always issue the readback, not just if it's requested.

- If we get the value (on some backends, enqueue the readback) on submit: the promise will eventually resolve itself and the value can be held onto by the promise. If so, steps should be issued from GPUQueue.submit() which cause the promise to eventually resolve (so the spec text for that readback needs to be in, or referenced from, the submit() spec text).
- If we get the value (on some backends, enqueue the readback) lazily on the first access to `.executionTime`: on some backends, the value has to be persistently allocated in an extra GPU memory suballocation, until the readback occurs. Seems more complex to me.

The behavior will be only very slightly different, because in the latter, no matter how long you wait, executionTime will never be resolved already when you get it. I don't think this difference is important.

Reply:
Will you make this change, or would you like to defer it for later? If you'd like to defer it, please add an `Issue: define exactly when this promise is resolved and whether the value is fetched on submit or lazily.`

Reply:
OK, I will add an `Issue` for it now. I consider that maybe we also can remove the promise if we resolve the value from GPUQueue.submit() because we have read the value from GPU buffer or command buffer (on Metal) which is asynchronous.

Reply:
Even if the readback is done greedily, it can't happen until all of the submit()ted work has completed, which is asynchronous.

Reply:
Remove the issue and clear that the value of execution time is fetched when GPUQueue.submit().

Reply:
That wasn't the only issue; we also need to actually specify in the definition of GPUQueue.submit() that it issues the work and resolves the promise. I'll add a TODO.